todo_app/app.py

```diff
@@ -1,57 +1,159 @@
-import sys
+from os import error
 from flask import Flask, render_template, request, redirect
 from todo_app.flask_config import Config
-from todo_app.data.session_items import *
+from todo_app.trelloclient import *
+
 
 app = Flask(__name__)
 app.config.from_object(Config)
 
-#error handling for 404
+status_mapping = []
+board_Name = "ToDo"
+board_Id = ""
+
+def build_status_mapping():
+    trello_Board = TrelloBoard()
+    result = trello_Board.get_BoardList()
+    if result.status_code == 200:
+        for item in result.json():
+            if item['name'] == board_Name:
+                board_Id = item['id']   
+    else:
+        return render_template("404.html", error="Trello board 'ToDo' not found!")
+
+    if not (board_Id is None):
+        trello_List = TrelloList()
+        lists = trello_List.get_List(id=board_Id)
+        if lists.status_code == 200:
+            for list in lists.json():
+                if list['name'] == 'To Do':
+                    status_mapping.append ({
+                        'id' : list['id'],
+                        'status' : "Not Started"
+                    })
+                elif list['name'] == 'Doing':
+                    status_mapping.append ({
+                        'id' : list['id'],
+                        'status' : "In Progress"
+                    })
+                elif list['name'] == 'Done':
+                    status_mapping.append ({
+                        'id' : list['id'],
+                        'status' : "Completed"
+                    })
+                else:
+                    pass
+                    #do nothing
+        else:
+            return render_template("404.html", error="problem connecting to Trello API endpoint")           
+    else:
+        return render_template("404.html", error="Trello board 'ToDo' not found!") 
```

Comment:
You're rendering these templates inside a general function call rather than a flask route, so this happens when the server starts rather than when a user hits the server, so it doesn't make much sense to render a template and it will error. Instead we might just want to log an error, or even crash the program

Reply:
Note sure on this one. I will setup a call with you to discuss this further.